File: 1474-delete-n-nodes-after-m-nodes-of-a-linked-list/1474-delete-n-nodes-after-m-nodes-of-a-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* deleteNodes(ListNode* head, int m, int n) {
        ListNode *sentinal = new ListNode(INT_MIN, head);
        ListNode *cur = head;
        ListNode *prev = sentinal;
        
        while( cur ) {
            for( int i = 0; i < m && cur; i++) {
                prev = cur;
                cur = cur->next;
            }
            
            for ( int i = 0; i < n && cur; i++) {
                ListNode *temp = cur;
                cur = cur->next;
                delete temp;
            }    
            prev->next = cur;
        }
        
        head = sentinal->next;
        return head;
    
    }
};

File: 1474-delete-n-nodes-after-m-nodes-of-a-linked-list/1474-delete-n-nodes-after-m-nodes-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteNodes(self, head, m, n):
        """
        :type head: ListNode
        :type m: int
        :type n: int
        :rtype: ListNode
        """
        sentinel = ListNode(0, head)
        cur = head
        prev = sentinel
        
        while cur is not None:
            i = 1
            while cur is not None and i <= m:
                prev = cur
                cur = cur.next
                i += 1
                
            if cur is None:
                break;
                
            j = 1
            while cur is not None and j <= n:
                cur = cur.next
                j += 1
            
            prev.next = cur
        
        return sentinel.next
    
        